### DISCUSSION TITLE
Nullable Enums fail validation

### DISCUSSION THREAD

**Discussion by leifniem** [colinhacks/zod#2757] [Q&A] [answered]
It seems to me that the enum validation happens first and therefore `undefined` and `null` fail this test, eventhough `.nullish()` is appended to the definition.

I sadly do not have the time to investigate this and it seems adding `undefined` to the enum itself is not possible. I'll try my luck with a union.

Thank you for all the work on this library and sorry for the sparse information.

**Comment by NextThread** [comment]
hey, I'd like to work on this issue

**Comment by JacobWeisenburger** [comment, marked as answer]
Everthing seems to be working as expected:

```ts
const schema = z.enum( [ 'a', 'b', 'c' ] ).nullable()
console.log( schema.parse( 'a' ) ) // 'a'
console.log( schema.parse( null ) ) // null
```

Please let me know if there is something I am missing.

**Comment by leifniem** [comment]
I don't know if i'll be able to reconstruct the specific case that got me this behaviour, since it happened during my bachelor thesis and the code changed a lot since then. I will close this for that matter, sorry if this was only an error on my side.
